clients/java/src/main/java/io/camunda/zeebe/client/impl/response/DocumentMetadataImpl.java

```diff
@@ -17,10 +17,21 @@
 
 import io.camunda.zeebe.client.api.response.DocumentMetadata;
 import java.time.ZonedDateTime;
+import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
 
 public class DocumentMetadataImpl implements DocumentMetadata {
 
+  private static final String EXPIRES_AT =
+      io.camunda.zeebe.client.protocol.rest.DocumentMetadata.JSON_PROPERTY_EXPIRES_AT;
+  private static final String SIZE =
+      io.camunda.zeebe.client.protocol.rest.DocumentMetadata.JSON_PROPERTY_SIZE;
+  private static final String FILE_NAME =
+      io.camunda.zeebe.client.protocol.rest.DocumentMetadata.JSON_PROPERTY_FILE_NAME;
+  private static final String CONTENT_TYPE =
+      io.camunda.zeebe.client.protocol.rest.DocumentMetadata.JSON_PROPERTY_CONTENT_TYPE;
```

Comment:
❓ Any specific reason for this ? Can we simply use the variables from `DocumentMetadata` ?

Reply:
I added this because I encountered problems with how additional properties are handled in the generated classes. 

`DocumentMetadata` is generated from the OpenAPI spec. This class allows arbitrary additional properties, and the generator produces an extension of `HashMap` with some custom fields and getters/setters. When we deserialize the class with Jackson, it always tries to resolve the hash map properties first and all the times I ran this my custom setters returned nulls, and `contentType` and other fields ended up in the HashMap as regular entries.

To make things worse, I didn't have such an issue on the server side because we are using different code generators for the client and for the server (`JavaCodegen` in the client, `SpringCodegen` in the gateway). `SpringCodegen` apparently uses a different generation strategy and doesn't produce a subclass of `HashMap` - so the properties are resolved correctly there.

I think this might be a bug in the `JavaCodegen`, but I'm not sure if we can rely on this behavior so I tried to handle both options - so that it doesn't break if this behavior changes.

This also applies to the ❓ above about `Long` vs `Integer` - hash map entries are deserialized as integer by default.

Reply:
Why you need `additionalProperties` ? Can you simply use the properties that you define in the specification ?
It might be a stupid question but I'm missing the reason

Reply:
We want to allow the users to store any arbitrary properties in the metadata, along with a set of pre-defined ones like contentType etc.

Reply:
Maybe to avoid these weird transformations we could deserialize the metadata as a `HashMap` (i.e. remove the dependency on the generated class). Then `DocumentMetadataImpl` could be a simple wrapper on top of the map where methods like `getContentType()` can simply return the value from the map using a predefined key. WDYT?

Reply:
In this way users can send every type of value for the predefined key, right ? 
e.g suppose to have a defined key SIZE, users can set for this key a String, when we are going to get this value from the map we should handle all this edge cases, I don't like it tbh.

What if in the spec you define the `DocumentMetadata` with the pre-defined property and then you add a new field `customProperties` that has a reference to a sub-object with only additional properties.

```
{
   size: 1,
   expireAt: "",
   fileName: "filename",
   contentType: "type",
   customProperties: {
      "foo": "bar"
   }
}
```

Reply:
I agree this is a potential flaw and I think this approach could be a solution. We already assumed the previous API design everywhere so I'll check if there are any issues with changing it at this stage. But in general I think it will work 👍

Reply:
Moved all custom field to `customProperties`